src/uavariant.cpp

```diff
@@ -488,24 +488,112 @@ UaString UaVariant::toFullString() const
-    return OpcUa_Good;
+    return convertNumericType(out);
+}
+
+template<typename TTargetNumericType>
+  UaStatus UaVariant::convertNumericType(TTargetNumericType* out ) const
+{
+	try
+	{
+		switch(m_impl->type->typeIndex)
+		{
+			case UA_TYPES_BOOLEAN:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Boolean*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_BYTE:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Byte*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_SBYTE:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_SByte*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_INT16:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Int16*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_UINT16:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_UInt16*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_INT32:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Int32*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_UINT32:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_UInt32*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_INT64:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Int64*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_UINT64:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_UInt64*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_FLOAT:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Float*>(m_impl->data)) );
+				return OpcUa_Good;
+			case UA_TYPES_DOUBLE:
+				*out = boost::numeric_cast<TTargetNumericType>( *(static_cast<OpcUa_Double*>(m_impl->data)) );
+				return OpcUa_Good;
+			default:
+				LOG(Log::DBG) << __FUNCTION__ << " conversion failed, no conversion handler for internal variant type ["<<m_impl->type->typeName<<"]";
+				return OpcUa_Bad;
+		}
+	}
+	catch(const boost::numeric::bad_numeric_cast& e)
+	{
+		LOG(Log::DBG) << __FUNCTION__ << " conversion failed for internal variant type ["<<m_impl->type->typeName<<"], boost conversion threw exception: " << e.what();
+		return OpcUa_BadOutOfRange;
+	}
+	catch(const std::runtime_error& e)
+	{
+		LOG(Log::DBG) << __FUNCTION__ << " conversion failed for internal variant type ["<<m_impl->type->typeName<<"], conversion threw exception: " << e.what();
+		return OpcUa_Bad;
+	}
+}
+
+bool UaVariant::isNumericType( const UA_DataType& dataType ) const
+{
+	switch(dataType.typeIndex)
+	{
+	case UA_TYPES_BOOLEAN:
+	case UA_TYPES_SBYTE:
+	case UA_TYPES_BYTE:
+	case UA_TYPES_INT16:
+	case UA_TYPES_UINT16:
+	case UA_TYPES_INT32:
+	case UA_TYPES_UINT32:
+	case UA_TYPES_INT64:
+	case UA_TYPES_UINT64:
+	case UA_TYPES_FLOAT:
+	case UA_TYPES_DOUBLE:
+		return true;
+	default:
+		return false;
+	}
 }
 
 template<typename T, typename U>
```

Comment:
What about support for arrays?

Reply:
Still no support - it's not worse than it was.

Reply:
OK but is this planned, or?